### PR TITLE
3.5.1 Educate to use Text over String via a notice.

### PR DESCRIPTION
**Pull request by inoas**
This should give library authors some more incentive/time to switch things around, for instance: https://github.com/WyriHaximus/TwigView/issues/52

@WyriHaximus please state if this would have helped you in any way, if not then this can be closed.

### CODE REVIEW

File: src/Utility/String.php [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <?php
 // @deprecated Backward compatibility with 2.x series
-if (PHP_VERSION_ID < 70000) {
```
Comment (dereuromark) [inline src/Utility/String.php]: the condition needs to stay.

**Comment by dereuromark** [comment]
I suggest pointing this to 3.next (3.6).

File: src/Utility/String.php (hunk 1 above)
Comment (WyriHaximus) [inline src/Utility/String.php]: Yup otherwise it blows on PHP 7

**Comment by WyriHaximus** [comment]
@inoas this would work just fine 👍

File: src/Utility/String.php (hunk 1 above)
Comment (inoas) [inline src/Utility/String.php]: Why? If you got < 7.0 you get no error.
If you use >= 7.0.0 you will either get an error if you do not alias the class (use Cake\Utility\String as CakeString) - or - if you alias the class - you will now get the notice... anyway I will re-add it. Just not sure it is required.

**Comment by inoas** [comment]
But why should it point to 3.6? This should have been here in 3.5.0 I think.
https://github.com/cakephp/cakephp/wiki/3.6-Roadmap

> In order to ease the migration into 4.x, all current deprecations that exist only as documentation warnings will be upgraded to run-time errors

So it should say "upgraded to run-time notices `E_USER_NOTICE`" `?

**Comment by dereuromark** [comment]
Deprecations are usually E_USER_DEPRECATED.
If this breaks hard as suggested, I agree that having it in 3.5 already as a higher run-time error.

**Comment by inoas** [comment]
Right now as far as I understood E_USER_NOTICE it will run, just in debug / in your log file, show errors. That should help until 3.6 hits.

**Comment by markstory** [comment]
@inoas We'll be using `E_USER_DEPRECATED` as that is the appropriate error level for deprecation warnings.

**Comment by inoas** [comment]
Well, let me know if I should reopen this against 3.next

**Pull request closed without merging**